### PR TITLE
Feature/login page

### PR DESCRIPTION
**Pull request by elizhahn**
# Is this a fix or a feature?
-feature
# What is the change?
- this adds a login page for a user to login with a specific username and password
- Error validation was created for the login page
- Error validation was created for the fetch request
# What does it fix?
# Where should the reviewer start?
- index.js: lines: 31 - 36, 43, 65-111
- index.html: lines: 10-22
# How should this be tested?
- clone down the repo, fetch the feature/login-page branch
- run npm start and visit localhost8080
- run npm start from the overlook api
*UI TESTING*: 
- login to the page with any password that does not match "customer30, customer21..etc"
- login to the page using the correct username of "customer5", but with no password
- login to the page using correct username "customer5" and correct password "overlook2021"
- You should have seen proper error messages, and upon using the correct username and password, you should be taken to the dashboard and the user's information should be displayed. 
- please review the UX and let me know if you think anything is missing or needs improvement. 
- please review the validation functions for areas where I can DRY it up

# Screenshots (if appropriate):

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -51,15 +62,58 @@ mainPage.addEventListener('click', updateAria);
...
+
+  function findUser(id) {
+    console.log(id[0])
+    fetch(`http://localhost:3001/api/v1/customers/${id[0]}`)
+    .then(response => checkForLoginError(response))
+    .then(data => {
+       console.log(data)
+       login(data, hotel.bookings);
+    })
+    .catch(err => displayServerError())
```
Comment (alexthompson207) [inline src/index.js:89]: Good error handling here!

File: src/index.js [hunk 2]
```diff
@@ -51,15 +62,58 @@ mainPage.addEventListener('click', updateAria);
...
   }
 
-  function displayUserInfo(roomData) {
+  function displayServerError() {
```
Comment (alexthompson207) [inline src/index.js:92]: Also here!

File: src/index.js [hunk 3]
```diff
@@ -51,15 +62,58 @@ mainPage.addEventListener('click', updateAria);
...
+  } else { 
+    hide(passwordErrorMsg);
+    const userId = inputUserName.value.match(/\d+/g)
+    findUser(userId)
+  }
+  }
+
+  function findUser(id) {
+    console.log(id[0])
+    fetch(`http://localhost:3001/api/v1/customers/${id[0]}`)
```
Comment (alexthompson207) [inline src/index.js:83]: Not sure how to implement this, but make sure this is dynamic....or I think?

**Review by alexthompson207 — COMMENTED** [review]
Your website looks fantastic! Totally reminds me of a real hotels site, great job!

Only spent a little time messing around on the hosted site, but when a user clicks on book a room, do you want it to update their booking history? Because I don't think that was happening to me

**Comment by elizhahn** [comment]
Yes good catch, that is on the list to do. I just made a card for it so I don't forget!

**Pull request merged**